### PR TITLE
Improve DataContainer I/O

### PR DESCRIPTION
**Pull request by PabloCorcho**
This PR implements the full set I/O methods to write and read data ingested by PyKOALA in an homogeneous way.

### CODE REVIEW

**Comment by PabloCorcho** [comment]
This includes a few new additions to the data model such as the use of `WCS` information at all DC levels.

**Comment by PabloCorcho** [comment]
I guess it would be nice if we can implement a few tests to prevent data corruption

**Review by miguelglezb — APPROVED** [review]

File: tutorials/koala_tutorial_1_intro_n.ipynb
Comment (paranoya) [inline tutorials/koala_tutorial_1_intro_n.ipynb:1]: Empty?

File: src/pykoala/data_container.py [hunk 1]
```diff
@@ -591,13 +724,16 @@ def __init__(self, **kwargs):
...
-            self.vprint(
-                "WARNING: No `wavelength` vector supplied; creating empty `SpectraContainer`"
-            )
-            self._wavelength = u.Quantity([], u.AA)
+            # TODO: re-implement with units refactoring
+            # self.vprint(
+            #     "WARNING: No `wavelength` vector supplied; creating empty `SpectraContainer`"
+            # )
+            #self._wavelength = u.Quantity([], u.AA)
+            self._wavelength = None
```
Comment (paranoya) [inline src/pykoala/data_container.py:735]: Does this make any sense?

File: tutorials/koala_tutorial_1_intro_n.ipynb
Comment (PabloCorcho) [reply tutorials/koala_tutorial_1_intro_n.ipynb:1]: Ops

File: tutorials/astrometry_offset_correction.ipynb
Comment (paranoya) [inline tutorials/astrometry_offset_correction.ipynb:1]: Clean output?

**Review by paranoya — APPROVED** [review]

**Pull request merged**